### PR TITLE
HDDS-3588. Fix NPE while getPipelines if absent in query2OpenPipelines

### PR DESCRIPTION
**Pull request by maobaolong**
## What changes were proposed in this pull request?

As title described, there maybe a NPE if get pipeline from query2OpenPipelines return null. It is possible if we can specify a replication number other than 1 and 3.

## What is the link to the Apache JIRA 

https://issues.apache.org/jira/browse/HDDS-3588

## How was this patch tested?

Now, this issue cannot be trigger until HDDS-3581 merged.

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateMap.java [hunk 1]
```diff
@@ -210,6 +210,11 @@ Pipeline getPipeline(PipelineID pipelineID) throws PipelineNotFoundException {
     Preconditions.checkNotNull(state, "Pipeline state cannot be null");
 
     if (state == PipelineState.OPEN) {
+      List<Pipeline> pipeline =
+          query2OpenPipelines.get(new PipelineQuery(type, factor));
```
Comment (avijayanhwx) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateMap.java:214]: Can we use getOrDefault here?

Comment (maobaolong) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateMap.java]: Thank you, done

**Comment by maobaolong** [comment]
@avijayanhwx Thank you for your review, i have addressed your comments, PTAL.

**Comment by ChenSammi** [comment]
+1.  Thanks @avijayanhwx for the review and @maobaolong  for fix the issue.

**Pull request merged**